src/timer/timer_enable.cc

```diff
@@ -215,97 +153,66 @@ void TimerStatCollector::Report(std::size_t index, cr_duration_nsec_t duration)
         }
     }
 
-    constexpr auto kDurationMask = (1ull << data_to_report.stat.kTotalDurationBits) - 1;
-
-    data_to_report.stat.hits              = 1;
-    data_to_report.stat.total_duration_ns = static_cast<std::uint64_t>(duration) & kDurationMask;
-    collector_entries_[index].duration_buckets_[bucket_idx].hit_and_total_duration_.fetch_add(data_to_report.raw);
-}
-
-void TimerStatCollector::Clear() {
-    for (auto&& entry : collector_entries_) {
-        for (auto&& bucket : entry.duration_buckets_) {
-            bucket.hit_and_total_duration_.store(0);
-        }
+    // If another thread is collecting data, or it happens to be a collision in reporting,
```

Comment:
This sounds very counter-intuitive and unreliable

Reply:
We do that because we don't want to wait for the locks. Since the data collection happens rarely, this should be ok.

As for collision, will use thread-local collectors instead, then there will be no collision.

Reply:
If It's true could you comment this assumption in? About the rarely part.

Reply:
Will do